source/common/router/retry_state_impl.cc

```diff
@@ -240,6 +243,20 @@ RetryStatus RetryStateImpl::shouldRetry(bool would_retry, DoRetryCallback callba
   return RetryStatus::Yes;
 }
 
+bool RetryStateImpl::retryBudgetExceeded() {
+  ASSERT(retry_budget_);
+
+  const uint64_t current_active = cluster_.resourceManager(priority_).requests().count() +
+                                  cluster_.resourceManager(priority_).pendingRequests().count();
+
+  if (current_active < retry_budget_->min_concurrency) {
```

Comment:
Don't you need to do the % calculation before doing this check otherwise you are not going to get the 15 * 0.2 = 3 behavior we talked about?

Reply:
I changed the way this all works in the coming patch. It's all much simpler now.